cmd/clusterctl/client/tree/discovery.go

```diff
@@ -69,10 +70,14 @@ func Discovery(ctx context.Context, c client.Client, namespace, name string, opt
 	}
 
 	// Adds control plane
-	controlPLane, err := external.Get(ctx, c, cluster.Spec.ControlPlaneRef, cluster.Namespace)
-	if err == nil {
-		tree.Add(cluster, controlPLane, ObjectMetaName("ControlPlane"), GroupingObject(true))
+	if cluster.Spec.ControlPlaneRef == nil {
+		return nil, errors.Errorf("controlplane ref cannot be nil")
```

Comment:
I think it would be preferable to use `New` over `Errorf` here since this isn't a format string
```suggestion
		return nil, errors.New("controlplane ref cannot be nil")
```